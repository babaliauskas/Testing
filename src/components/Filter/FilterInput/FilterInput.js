import React from 'react';
import PropTypes from 'prop-types';

const FilterInput = ({ handleFilter, filter, handleSortPizzas }) => (
  <div className="FilterInput">
    <input type="text" onChange={event => handleFilter(event)} value={filter} />
    <button className="btn-sort" type="button" onClick={handleSortPizzas}>Sort Pizzas</button>
  </div>
);
FilterInput.propTypes = {
  handleFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
  handleSortPizzas: PropTypes.func.isRequired,
};

export default FilterInput;
