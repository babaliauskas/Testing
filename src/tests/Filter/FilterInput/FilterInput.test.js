/* global it, describe, expect, jest, beforeEach */
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FilterInput from '../../../components/Filter/FilterInput/FilterInput';

configure({ adapter: new Adapter() });

describe('FilterInput', () => {
  const mockFilter = jest.fn();
  const mockSortPizzas = jest.fn();
  const props = {
    handleFilter: mockFilter,
    filter: '',
    handleSortPizzas: mockSortPizzas,
  };
  const filterinput = shallow(<FilterInput {...props} />);

  it('should render properly', () => {
    expect(filterinput).toMatchSnapshot();
  });

  it('should contain input field', () => {
    expect(filterinput.find('input').exists()).toBe(true);
  });

  it('should contain button ', () => {
    expect(filterinput.find('button').exists()).toBe(true);
  });

  describe('Input field', () => {
    const inputChange = 'sa';
    beforeEach(() => {
      filterinput.find('input').simulate('change', inputChange);
    });

    it('should contain the right input value ', () => {
      expect(mockFilter).toHaveBeenCalledWith(inputChange);
    });
  });
});
