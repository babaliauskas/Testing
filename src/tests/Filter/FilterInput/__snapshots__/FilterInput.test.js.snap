// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterInput should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterInput
    filter=""
    handleFilter={[MockFunction]}
    handleSortPizzas={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          type="text"
          value=""
        />,
        <button
          className="btn-sort"
          onClick={[MockFunction]}
          type="button"
        >
          Sort Pizzas
        </button>,
      ],
      "className": "FilterInput",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sort Pizzas",
          "className": "btn-sort",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "Sort Pizzas",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            type="text"
            value=""
          />,
          <button
            className="btn-sort"
            onClick={[MockFunction]}
            type="button"
          >
            Sort Pizzas
          </button>,
        ],
        "className": "FilterInput",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sort Pizzas",
            "className": "btn-sort",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "Sort Pizzas",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
